src/app.js

```diff
@@ -204,14 +327,223 @@ gpii.app.performQuit = function () {
-            gpii.rejectToLog(that.keyOut(that.model.keyedInUserToken), "Couldn't logout current user"),
+            gpii.rejectToLog(that.keyOut(), "Couldn't logout current user"),
             gpii.app.performQuit
         ]);
     } else {
         gpii.app.performQuit();
     }
 };
 
+fluid.registerNamespace("gpii.app.pcp");
+
+/**
+ * Sends a message to the given window
+ *
+ * @param pcpWindow {Object} An Electron `BrowserWindow` object
+ * @param messageChannel {String} The channel to which the message to be sent
+ * @param message {String}
+ */
+gpii.app.pcp.notifyPCPWindow = function (pcpWindow, messageChannel, message) {
+    if (pcpWindow) {
+        pcpWindow.webContents.send(messageChannel, message);
+    }
+};
+
+/**
+ * Creates an Electron `BrowserWindow` that is to be used as the PCP window
+ *
+ * @param width {Number} The desired width of the window
+ * @param height {Number} The desired height of the window
+ * @returns {Object} The created Electron `BrowserWindow`
+ */
+gpii.app.pcp.makePCPWindow = function (width, height) {
+    var screenSize = electron.screen.getPrimaryDisplay().workAreaSize;
+    // TODO Make window size relative to the screen size
+    var pcpWindow = new BrowserWindow({
+        width: width,
+        height: height,
+        show: false,
+        frame: false,
+        fullscreenable: false,
+        resizable: false,
+        alwaysOnTop: true,
+        skipTaskbar: true,
+        x: screenSize.width - width,
+        y: screenSize.height - height,
+
+        // Make the app feel more native
+        backgroundColor: "#71c5ef"
+    });
+
+    pcpWindow.on("blur", function () {
+        pcpWindow.hide();
+    });
+
+    var url = fluid.stringTemplate("file://%dirName/pcp/index.html", {
+        dirName: __dirname
+    });
+    pcpWindow.loadURL(url);
+
+    return pcpWindow;
+};
+
+/**
+ * Shows the passed Electron `BrowserWindow`.
+ * In case the window is already opened, it is focused.
+ *
+ * @param pcpWindow {Object} An Electron `BrowserWindow`.
+ */
+gpii.app.pcp.showPCPWindow = function (pcpWindow) {
+    if (!pcpWindow.isVisible()) {
+        pcpWindow.show();
+    }
+
+    pcpWindow.focus();
+};
+
+/**
+ * Creates a setting view model to be used in the settings window.
+ * @param key {String} The name of the setting. Must be unique as
+ * subsequent requests to the GPII API will use this key as identifier.
+ * @param settingDescriptor {Object} A descriptor for the given setting
+ * containing its title, description and constraints regarding its value.
+ * @return {Object} The view model for the setting.
+ */
+gpii.app.createSettingModel = function (key, settingDescriptor) {
+    var schema = settingDescriptor.schema;
+
+    return {
+        path: key,
+        value: settingDescriptor.value,
+        solutionName: settingDescriptor.solutionName,
+
+        title: schema.title,
+        description: schema.description,
+        icon: "../icons/gear-cloud-black.png",
```

Comment:
Hoist this out into a component option

Reply:
Sorry, ignore this comment, I see this is a placeholder for upcoming functionality - just add a TODO comment here to clarify this